plugins/aws_ecr/lib/samson_aws_ecr/samson_plugin.rb

```diff
@@ -0,0 +1,21 @@
+require 'aws-sdk'
+
+module SamsonAwsEcr
+  class Engine < Rails::Engine
+    mattr_accessor :ecr_client do
+      Aws::ECR::Client.new(region: ENV['AWS_REGION'] || "us-west-2")
+    end
+  end
+end
+
+Samson::Hooks.callback :before_docker_build do
+  begin
+    authorization_token = SamsonAwsEcr::Engine.ecr_client.get_authorization_token
+    authorization_data  = authorization_token.authorization_data.first
+    user, pass = Base64.decode64(authorization_data.authorization_token).split(":")
+    ENV['DOCKER_REGISTRY_USER'] = user
```

Comment:
Also, please only run this code if `ENV['DOCKER_REGISTRY']` matches a regex for an ECR registry.

I frequently run Samson with `PLUGINS=all`, and I don't want it to try to authenticate with ECR if I'm not pointing to an ECR registry.